internal/gps/source_manager_test.go

```diff
@@ -13,71 +13,82 @@ import (
 
 func TestSourceManager_InferConstraint(t *testing.T) {
 	t.Parallel()
-	h := test.NewHelper(t)
-	cacheDir := "gps-repocache"
-	h.TempDir(cacheDir)
-	sm, err := NewSourceManager(h.Path(cacheDir))
-	h.Must(err)
-
-	sv, err := NewSemverConstraintIC("v0.8.1")
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	svs, err := NewSemverConstraintIC("v0.12.0-12-de4dcafe0")
-	if err != nil {
-		t.Fatal(err)
-	}
+	testcase := func(str string, pi ProjectIdentifier, want Constraint) func(*testing.T) {
```

Comment:
nit: Can these subtests be restructured to match the way we have been doing other subtests? The pattern that we have been following is to create an array or map of testcase data, and then in a for loop call `t.Run`. It makes it easier to review when all the testcases are together.

https://github.com/golang/dep/blob/6c5ebb93c4e32614ba61fed6acb962a272ad4b8a/context_test.go#L116-L128

Reply:
Sure, done.